### PR TITLE
Feat/member-validation

### PR DESCRIPTION
**Pull request by ricksanchez**
First part of a series that will allow to integrate results with problem details and standalone json conversions of errors, the goal is to have practical parsing/validation for input objects and report back the errors in their members.

### CODE REVIEW

**Review by mtyski — COMMENTED** [review]
Please add conditional expressions that allow AOT-compatibility (if that is a target for us in your opinion):

```
  <PropertyGroup Condition="'$(TargetFramework)' == 'net8.0'">
    <IsAotCompatible>true</IsAotCompatible>
  </PropertyGroup>
```

There are a couple of warnings in the JSON serializers, which can be worked around through System.Text.Json code generation.

**Comment by ricksanchez** [comment]
> Please add conditional expressions that allow AOT-compatibility (if that is a target for us in your opinion):
> 
> ```
>   <PropertyGroup Condition="'$(TargetFramework)' == 'net8.0'">
>     <IsAotCompatible>true</IsAotCompatible>
>   </PropertyGroup>
> ```
> 
> There are a couple of warnings in the JSON serializers, which can be worked around through System.Text.Json code generation.

Everything was moved to a new project, to preserve AoT for the main project

File: src/Results.Immutable.Tests/MemberTests/MemberParserTests.cs [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+                roads > 3,
+                roads,
+                new Error("Not enough")),
+            propertyNamingPolicy: JsonNamingPolicy.CamelCase);
+
+        result.Errors.Should()
+            .ContainSingle()
+            .Which.Should()
+            .Match<MemberError>(x => x.Member == "coVariant");
+    }
```
Comment (mtyski) [inline src/Results.Immutable.Tests/MemberTests/MemberParserTests.cs:150]:
```suggestion
        result.Errors.Should()
            .SatisfyRespectively(
                static single =>
                    single.Should()
                        .Match<MemberError>(
                            static x => x.Member == "coVariant"));
    }
```

File: src/Results.Immutable.Tests/MemberTests/MemberParserTests.cs [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+                        nameof(Command.Age),
+                        "The user should be an adult",
+                        ImmutableList.Create(errorTooYoung)),
+                    new MemberError(
+                        nameof(Command.Name),
+                        "",
+                        ImmutableList.Create(errorTooShort))));
+    }
+
+    public async Task ValidateAsynchronouslyValuesWithoutErrors()
```
[the remark this answers is not in the document]
Comment (ricksanchez) [reply src/Results.Immutable.Tests/MemberTests/MemberParserTests.cs:67]: done

File: src/Results.Immutable.Tests/MemberTests/MemberParserTests.cs [hunk 3]
```diff
@@ -0,0 +1,200 @@
+using System.Linq.Expressions;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Results.Immutable.Member;
+using static Results.Immutable.Result;
+
+namespace Results.Immutable.Tests.MemberTests;
+
+public class MemberParserTests
```
[the remark this answers is not in the document]
Comment (ricksanchez) [reply src/Results.Immutable.Tests/MemberTests/MemberParserTests.cs:9]: done

File: yarn.lock [hunk 4]
```diff
@@ -919,6 +919,14 @@ __metadata:
...
   linkType: soft
 
+"Results.Immutable.Members@workspace:src/Results.Immutable.Members":
+  version: 0.0.0-use.local
+  resolution: "Results.Immutable.Members@workspace:src/Results.Immutable.Members"
+  dependencies:
+    Results.Immutable: ^1.0.0-alpha.1
+  languageName: unknown
+  linkType: soft
+
```
[the remark this answers is not in the document]
Comment (ricksanchez) [reply yarn.lock:929]: right, deleted

File: src/Results.Immutable.Tests/MemberTests/MemberParserTests.cs (hunk 1 above)
Comment (ricksanchez) [reply src/Results.Immutable.Tests/MemberTests/MemberParserTests.cs]: done

**Review by mtyski — APPROVED** [review]
LGTM!

**Pull request merged**